### PR TITLE
Support for generic yaml insertion in Enterprise

### PR DESCRIPTION
**Pull request by jmunozro**
# Description

Adding support for generic yaml insertion in Enterprise

# Context

[Related issue](https://github.com/solo-io/gloo/issues/5629)

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: install/helm/gloo/templates/18-settings.yaml [hunk 1]
```diff
@@ -117,11 +117,15 @@ spec:
 {{/* extauth have specified yaml in Enterprise (as opposed to generic yaml insertion for third party configs here) */}}
 {{- if .Values.global.extauthCustomYaml }}
 {{- if .Values.global.extensions }}
-{{- if .Values.global.extensions.extAuth }}
+{{- if or .Values.global.extauthCustomYamlBody .Values.global.extensions.extAuth }}
```
Comment (kdorosh) [inline install/helm/gloo/templates/18-settings.yaml:120]: can we do this already with the kube overrides? https://github.com/solo-io/gloo/blob/5fa02220bc55b63d653219b3f6c0b04fee4c7b60/install/helm/gloo/templates/18-settings.yaml#L150-L155

is this a common enough use case to support natively?

File: install/helm/gloo/templates/18-settings.yaml [hunk 2]
```diff
@@ -117,11 +117,15 @@ spec:
...
   extauth:
+{{- if .Values.global.extauthCustomYamlBody }}
```
Comment (sam-heilbron) [inline install/helm/gloo/templates/18-settings.yaml:122]: I'm still a little confused how exposing a new field, and injecting the yaml of both fields resolves the issue. Can you help me understand?

Comment (jmunozro) [reply install/helm/gloo/templates/18-settings.yaml:122]: Currently, `.Values.global.extensions.extAuth` is 'polluted' with deployment variables, like `.Values.global.extensions.extAuth.deployment`
Those entries are needed for the enterprise chart, but will make the OSS Settings break when `.Values.global.extauthCustomYaml` is true.

AFAIK, there is no way to 'overwrite' or manipulate the global values in a parent chart and then using the altered result for a child chart. Changing the `.Values.global.extensions.extAuth.deployment` and moving it to a proper place will cause an impact in many customers, so duplicating the variable seemed like the less dirty solution.

File: install/helm/gloo/templates/18-settings.yaml (hunk 1 above)
Comment (jmunozro) [reply install/helm/gloo/templates/18-settings.yaml:120]: Yep that could work. I'm putting this PR in draft and proposing the solution to the customer

THANKS!

**Comment by jmunozro** [comment]
https://github.com/solo-io/gloo/pull/5641#discussion_r758214491

**Pull request closed without merging**